db/migrate/20170301192129_create_provider_tags_table.rb

```diff
@@ -0,0 +1,11 @@
+class CreateProviderTagsTable < ActiveRecord::Migration[5.0]
+  def change
+    create_table :provider_tags do |t|
+      t.string :key, :null => false, :comment => "The key in a key/value pair."
+      t.string :value, :comment => "The value within a key/value pair."
+      t.string :resource_id, :null => false, :comment => "Reference to the field used as primary key of the resource."
+      t.string :type, :null => false, :comment => "The model name of the resource type."
```

Comment:
@blomquisg Maybe I'm behind the times on AR, but I thought you generally picked polymorphic or STI. Perhaps I can switch this to polymorphic, but I was worried about the potentially different primary key used for each object type.